### PR TITLE
Second attempt at a badges API

### PR DESCRIPTION
**Pull request by matthewjamesadam**
This is a WIP for getting feedback for #62783

Alternative API is here: https://github.com/microsoft/vscode/pull/139225

### CODE REVIEW

File: src/vscode-dts/vscode.proposed.badges.d.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+	export namespace window {
+		/**
+		 * Register a {@link ActivityProvider} for the view contributed using the extension point `views`.
+		 * This will allow you to contribute activity state for the view.
+		 *
+		 * @param viewId Id of the view contributed using the extension point `views`.
+		 * @param activityProvider A {@link ActivityProvider} that provides activity state for the view
+		 */
+		export function registerActivityProvider(viewId: string, activityProvider: ActivityProvider): Disposable;
```
Comment (matthewjamesadam) [inline src/vscode-dts/vscode.proposed.badges.d.ts:80]: It's unclear to me if the thing you would register the activity provider for would be the ID of the view, or the view container.  I believe you can contribute activity information for both, but I'm not sure if an extension API should support both or not.

**Comment by alexr00** [comment]
@matthewjamesadam thank you for the PRs! Just an heads up that our API process can be a long: https://github.com/Microsoft/vscode/wiki/Extension-API-process

I will bring your proposals to our next team API meeting and let you know the feedback in the PRs.

**Comment by matthewjamesadam** [comment]
Thank you @alexr00 , I appreciate that there is a process for changing the extension API and that it might take awhile.  Thanks for taking this to the API meeting, let me know if there is any other way I can help.

**Comment by matthewjamesadam** [comment]
@alexr00 just FYI, I have slightly amended this proposal to use classes instead of interfaces for the badge API, and I have implemented the API in another PR that branches off of this one: https://github.com/microsoft/vscode/pull/139539.

**Comment by alexr00** [comment]
Of the two API proposals, the other is closer to what we're looking for in badges! I've left my feedback there: https://github.com/microsoft/vscode/pull/139225#issuecomment-1015420619. Closing in favor of the other proposal.

**Comment by matthewjamesadam** [comment]
Thanks for the feedback @alexr00 !

**Pull request closed without merging**